rhel/repositoryscanner.go

```diff
@@ -119,16 +121,32 @@ func (r *RepositoryScanner) Scan(ctx context.Context, l *claircore.Layer) (repos
 	return repositories, nil
 }
 
-// ContentSets returns a slice of CPEs bound into strings, as discovered by
+// getEmbeddedContentSets returns a slice of content-sets bound into strings, as discovered by
 // examining information contained within the container.
-func contentSets(l *claircore.Layer) ([]string, error) {
-	// TODO: Get CPEs using embedded content-set files.
-	// The files will be stored most likely in /root/buildinfo/ and will need to
+func (r *RepositoryScanner) getEmbeddedContentSets(ctx context.Context, log *zerolog.Logger, l *claircore.Layer) ([]string, error) {
+	// Get CPEs using embedded content-set files.
+	// The files is be stored in /root/buildinfo/content_manifests/ and will need to
 	// be translated using mapping file provided by Red Hat's PST team.
-	return nil, nil
+	path, buf, err := findContentManifestFile(log, l)
+	switch {
+	case err == nil:
+	case buf == nil:
+		fallthrough
+	case errors.Is(err, claircore.ErrNotFound):
+		return nil, nil
+	default:
+		return nil, err
+	}
+	log.Debug().Str("manifest-path", path).Msg("Found content manifest file")
+	contentManifestData := contentmanifest.ContentManifest{}
+	err = json.NewDecoder(buf).Decode(&contentManifestData)
+	if err != nil {
+		return nil, err
+	}
+	return contentManifestData.ContentSets, nil
 }
 
-func (r *RepositoryScanner) containerAPI(ctx context.Context, log *zerolog.Logger, l *claircore.Layer) ([]string, error) {
+func (r *RepositoryScanner) getContentSetsUsingContainerAPI(ctx context.Context, log *zerolog.Logger, l *claircore.Layer) ([]string, error) {
```

Comment:
cc @hdonnay when airgap gets designed, this RepoScanner will need to be configured *NOT* to call this method if airgap is enabled.